hibernate-core/src/main/java/org/hibernate/dialect/DerbyDialect.java

```diff
@@ -440,9 +453,16 @@ public boolean supportsTupleDistinctCounts() {
 	}
 
 	protected SqlTypeDescriptor getSqlTypeDescriptorOverride(int sqlCode) {
-		return sqlCode == Types.NUMERIC
-				? DecimalTypeDescriptor.INSTANCE
-				: super.getSqlTypeDescriptorOverride(sqlCode);
+		switch ( sqlCode ) {
+			case Types.BOOLEAN:
+				return SmallIntTypeDescriptor.INSTANCE;
```

Comment:
Same as above: why was this necessary?

Reply:
Same rationale as for DB2 from above: https://github.com/hibernate/hibernate-orm/pull/3667#discussion_r531191675

Derby has no booleans

Reply:
Are you certain? https://db.apache.org/derby/docs/10.8/ref/rrefsqljBoolean.html